app/controllers/simplified-account/settings/organisation-details/edit-organisation-details.controller.js

```diff
@@ -39,6 +45,19 @@ function post (req, res) {
     })
   }
 
+  const serviceUpdates = new ServiceUpdateRequest()
+    .replace().merchantDetails.name(req.body.organisationName)
+    .replace().merchantDetails.addressLine1(req.body.addressLine1)
+    .replace().merchantDetails.addressLine2(req.body.addressLine2)
+    .replace().merchantDetails.addressCity(req.body.addressCity)
+    .replace().merchantDetails.addressPostcode(req.body.addressPostcode)
+    .replace().merchantDetails.addressCountry(req.body.addressCountry)
+    .replace().merchantDetails.telephoneNumber(req.body.telephoneNumber)
+    .replace().merchantDetails.url(req.body.organisationUrl)
```

Comment:
Oh interesting, do you have to call `.replace()` every time? Seems a bit weird. I'm coming at this from Java-land where I'd probably use the Builder pattern.

Reply:
This essentially is a builder but it uses function currying to specify the patch operations, instead of having methods like `replaceName()`, `replaceAddressLine1()` etc. and `addName()`, `addAddressLine1()` etc. this achieves the same effect without writing out all those functions

Reply:
Oh, that's neat. I should do this more often in Java.